src/main/java/com/ita/if103java/ims/dto/TransactionDto.java

```diff
@@ -0,0 +1,146 @@
+package com.ita.if103java.ims.dto;
+
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.ita.if103java.ims.entity.TransactionType;
+
+import java.math.BigInteger;
+import java.sql.Timestamp;
+
+import static com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL;
+
+public class TransactionDto {
+    private BigInteger id;
+    private Timestamp timestamp;
+    private TransactionType type;
+    private AccountDto accountDto;
+    private UserDto workerId;
+    private ItemDto itemDto;
+    private Long quantity;
+
+    @JsonInclude(NON_NULL)
+    private AssociateDto associateDto;
+    @JsonInclude(NON_NULL)
+    private WarehouseDto movedFrom;
+    @JsonInclude(NON_NULL)
+    private WarehouseDto movedTo;
+
+    public TransactionDto() {
+    }
+
+    public TransactionDto(BigInteger id, Timestamp timestamp, TransactionType type, AccountDto accountDto,
+                          UserDto workerId, ItemDto itemDto, Long quantity) {
+        this.id = id;
+        this.timestamp = timestamp;
+        this.type = type;
+        this.accountDto = accountDto;
+        this.workerId = workerId;
+        this.itemDto = itemDto;
+        this.quantity = quantity;
+    }
+
+    public TransactionDto(BigInteger id, Timestamp timestamp, TransactionType type, AccountDto accountDto,
+                          UserDto workerId, ItemDto itemDto, Long quantity, AssociateDto associateDto) {
+        this.id = id;
+        this.timestamp = timestamp;
+        this.type = type;
+        this.accountDto = accountDto;
+        this.workerId = workerId;
+        this.itemDto = itemDto;
+        this.quantity = quantity;
+        this.associateDto = associateDto;
+    }
+
+    public TransactionDto(BigInteger id, Timestamp timestamp, TransactionType type, AccountDto accountDto,
+                          UserDto workerId, ItemDto itemDto, Long quantity, WarehouseDto movedFrom, WarehouseDto movedTo) {
+        this.id = id;
+        this.timestamp = timestamp;
+        this.type = type;
+        this.accountDto = accountDto;
+        this.workerId = workerId;
+        this.itemDto = itemDto;
+        this.quantity = quantity;
+        this.movedFrom = movedFrom;
+        this.movedTo = movedTo;
+    }
```

Comment:
I think it's better to make constructors like Builder design patter. I've just made it in Event entity and dto (branch event-controller)

Reply:
Yep, I've checked, thanks
I'll think about that ;)